### PR TITLE
[Article] "Filament: What to Expect in 2024"

### PR DESCRIPTION
**Pull request by alexandersix**
This is a slightly more fleshed-out version of my notes from the Filament 2024 Kickoff call. Give it a look and make sure I'm conveying everything accurately!

### CODE REVIEW

**Review by awcodes — COMMENTED** [review]
Looks good to me. Good job.

File: content/articles/alexandersix-filament-2024-goals.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+slug: alexandersix-filament-what-to-expect-in-2024
+author_slug: alexandersix
+publish_date: 2024-01-03
+categories: [general]
+type_slug: news
+---
+
+![Filament: What to Expect in 2024 Banner Image](/images/content/articles/alexandersix-filament-what-to-expect-in-2024/filament-what-to-expect-in-2024.webp)
+
+2023 was an incredible year for Filament, and the momentum just keeps on going into 2024! The Filament core team has taken a lot of time over the past couple of months to develop our goals for 2024, and we're excited to share them with you here! This list was originally shared during the Filament 2024 Kickoff event, so if you want to be the first to year about our yearly goals, make sure to tune in to our Filament 2025 Kickoff event next year!
```
Comment (ralphjsmit) [inline content/articles/alexandersix-filament-2024-goals.md:12]:
```suggestion
2023 was an incredible year for Filament, and the momentum just keeps on going into 2024! The Filament core team has taken a lot of time over the past couple of months to develop our goals for 2024, and we're excited to share them with you here! This list was originally shared during the Filament 2024 Kickoff event, so if you want to be the first to hear about our yearly goals, make sure to tune in to our Filament 2025 Kickoff event next year!
```

File: content/articles/alexandersix-filament-2024-goals.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+#### Filament v4
+
+This one's a bit of a stretch goal for us, but a lot of the changes and ideas that we have for Filament would likely be best served in a new major version. Our intention with v3 was to keep it around for a long time, but there's a chance that we release our next major version at the end of 2024!
+
+### Filament Theming Goals
+
+#### Theming API Updates
+
+Currently, most everything that Filament's users are wanting to do in their themes can be done using the new APIs added to Filament components in v3. However, the implementation isn't as technically sound as we'd like. This is mostly down to the fact that, when our default styles are overridden to create themes, the original Tailwind classes are still sticking around in the class string. This has two big issues that we want to address:
```
Comment (ralphjsmit) [inline content/articles/alexandersix-filament-2024-goals.md:68]:
```suggestion
Currently, almost everything that Filament's users are wanting to do in their themes can be done using the new APIs added to Filament components in v3. However, the implementation isn't as technically sound as we'd like. This is mostly down to the fact that, when our default styles are overridden to create themes, the original Tailwind classes are still sticking around in the class string. This has two big issues that we want to address:
```

**Review by ralphjsmit — COMMENTED** [review]
Nice & comprehensive article!

**Comment by alexandersix** [comment]
> Just casually noticed this one in the list of open PRs, nice & comprehensive article!

Thank you for all your help with the summarization, @ralphjsmit!

File: content/articles/alexandersix-filament-2024-goals.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+2023 was, by practically all accounts, the best year for Filament so far. Lots of accomplishments all around by the core team and the community, but here are a few of our favorites:
+
+### Filament v3 & v3.1
+
+[Filament v3](https://filamentphp.com/community/danharrin-filament-v3) was released in August, following the very successful launch of Livewire v3! This release was followed up quickly by v3.1, our largest minor release to date (more information on v3.1 can be found [here](https://filamentphp.com/community/alexandersix-filament-v3-1)).
+
+### New Team Members
+
+The core team gained three new members in 2023: [Hassan](https://zahirnia.com/) to lead the UI/UX design for Filament, [Saade](https://github.com/saade) to revamp the plugin submission process, and [Alex](https://alexandersix.com) to head up Filament's developer relations. With the new additions, Filament's core team is now made up of eight members!
```
Comment (danharrin) [inline content/articles/alexandersix-filament-2024-goals.md:24]:
```suggestion
The core team gained three new members in 2023: [Hassan](https://zahirnia.com) to lead the UI/UX design for Filament, [Saade](https://github.com/saade) to revamp the plugin submission process, and [Alex](https://alexandersix.com) to head up Filament's developer relations. With the new additions, Filament's core team is now made up of eight members!
```

File: content/articles/alexandersix-filament-2024-goals.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+- **Reveal-able Password Inputs** – a small quality-of-life addition that allows passwords to be toggled visible within a password input.
+- **New Toggle Button UI** – an alternative UI component for radio buttons and checkboxes (more information [here](https://github.com/filamentphp/filament/pull/9860))
+- **Split Component for Forms** – adding the [Split](https://filamentphp.com/docs/3.x/infolists/layout/split) component from Infolists to Forms
+- **“Wait, Don't Leave!” Functionality** – an alert for the User if they attempt to leave a page before a form is submitted or while a confirmation modal is open
+- **Apply Button for Filters** – defers the processing of filter queries until the button is clicked, instead of running a filter query on each individual filter change
+- **CSV Exports** – companion to the [CSV imports](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/import) introduced in v3.1
+- **Block Groups for Builder Fields** – organization groups for Builder blocks instead of a single dropdown list for easier block discovery
```
Comment (danharrin) [inline content/articles/alexandersix-filament-2024-goals.md:50]:
```suggestion
```

File: content/articles/alexandersix-filament-2024-goals.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+1. Specificity – there are ways around this, but it's clunky to write long CSS selector strings or sprinkle `!important` throughout your codebase
+2. Payload size – with Livewire sending HTML over the wire, the more Tailwind classes you have in the DOM, the larger the payload that Livewire has to send, receive, and parse
+
+There are a few ideas on the table for how we solve this issue while maintaining discoverability of styles and an easy-to-use API, but we haven't settled on one just yet.
+
+#### Filament Component Explorer
+
+Filament has a *lot* of UI components that can be used in TALL-stack projects, and in 2024, we want to help developers discover, view, and integrate them into their projects!
+
+In the JavaScript world, [Storybook](https://storybook.js.org/) is a fantastic piece of software that helps developers view and use components within their projects. We want to create an experience similar to Storybook for our Filament UI components that allows easy copy/paste usage of the Blade components.
```
Comment (danharrin) [inline content/articles/alexandersix-filament-2024-goals.md:79]:
```suggestion
In the JavaScript world, [Storybook](https://storybook.js.org) is a fantastic piece of software that helps developers view and use components within their projects. We want to create an experience similar to Storybook for our Filament UI components that allows easy copy/paste usage of the Blade components.
```

**Review by danharrin — CHANGES_REQUESTED** [review]
Please upload the original image to content/article/images. The `optimize-images` script can be used to autogenerate the WEBP version

**Comment by danharrin** [comment]
ignore me lol

**Pull request merged**